File: backend/routes/textanalysis.js
const express = require('express');
const router = express.Router();
const Groq = require('groq-sdk');

const groq = new Groq({ apiKey: process.env.GROQ_API_KEY });

router.post('/analyze', async (req, res) => {
  const { text } = req.body;
  if (!text) return res.status(400).json({ error: 'No text provided' });

  try {
    const completion = await groq.chat.completions.create({
      model: 'llama-3.3-70b-versatile',
      messages: [
        {
          role: 'system',
          content: `You are an AI assistant specialized in Autism Spectrum Disorder (ASD) behavioral analysis. 
Always respond ONLY with valid JSON, no extra text, no markdown, no backticks.`
        },
        {
          role: 'user',
          content: `Analyze this text from an individual with ASD: "${text}"

Respond ONLY in this exact JSON format:
{
  "emotions": {
    "frustration": <number 0-100>,
    "anxiety": <number 0-100>,
    "happiness": <number 0-100>,
    "sadness": <number 0-100>,
    "calm": <number 0-100>
  },
  "dominant_emotion": "<single word>",
  "intent": "<one sentence describing what person is communicating>",
  "sarcasm_detected": <true or false>,
  "caregiver_suggestion": "<2 sentences of specific actionable advice for caregiver>",
  "alert_level": "<low, medium, or high>"
}`
        }
      ],
      temperature: 0.3,
      max_tokens: 500,
    });

    const raw = completion.choices[0].message.content.trim();
    const clean = raw.replace(/```json|```/g, '').trim();
    const parsed = JSON.parse(clean);
    res.json(parsed);

  } catch (err) {
    console.error('Groq error:', err.message);
    res.status(500).json({ error: 'Analysis failed', detail: err.message });
  }
});

module.exports = router;

File: flask-backend/routes/textanalysis.py
from flask import Blueprint, request, jsonify
import os
from groq import Groq

text_bp = Blueprint('text', __name__)
client = Groq(api_key=os.getenv('GROQ_API_KEY'))

@text_bp.route('/analyze', methods=['POST'])
def analyze():
    data = request.json
    text = data.get('text', '')
    if not text:
        return jsonify({'error': 'No text provided'}), 400
    try:
        completion = client.chat.completions.create(
            model='llama-3.3-70b-versatile',
            messages=[
                {'role': 'system', 'content': 'You are an ASD behavioral analysis AI. Always respond ONLY with valid JSON, no markdown, no backticks.'},
                {'role': 'user', 'content': f'''Analyze this text from an individual with ASD: "{text}"
Respond ONLY in this exact JSON:
{{"emotions":{{"frustration":<0-100>,"anxiety":<0-100>,"happiness":<0-100>,"sadness":<0-100>,"calm":<0-100>}},"dominant_emotion":"<word>","intent":"<one sentence>","sarcasm_detected":<true/false>,"caregiver_suggestion":"<2 sentences>","alert_level":"<low/medium/high>"}}'''}
            ],
            temperature=0.3, max_tokens=500
        )
        import json
        raw = completion.choices[0].message.content.strip()
        parsed = json.loads(raw.replace('```json','').replace('```','').strip())
        return jsonify(parsed)
    except Exception as e:
        return jsonify({'error': str(e)}), 500